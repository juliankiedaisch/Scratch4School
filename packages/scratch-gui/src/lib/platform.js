export const PLATFORM = {
    DESKTOP: 'DESKTOP',
    WEB: 'WEB',
    ANDROID: 'ANDROID'
};
