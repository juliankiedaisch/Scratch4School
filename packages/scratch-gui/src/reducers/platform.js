import {PLATFORM} from '../lib/platform.js';

const SET_PLATFORM = 'scratch-gui/Platform/SET_PLATFORM';

const initialState = {
    platform: PLATFORM.WEB
};

const reducer = function (state, action) {
    if (typeof state === 'undefined') state = initialState;
    switch (action.type) {
    case SET_PLATFORM:
        return {
            platform: action.platform
        };
    default:
        return state;
    }
};

const setPlatform = function (platform) {
    return {
        type: SET_PLATFORM,
        platform: platform
    };
};

export {
    reducer as default,
    initialState as platformInitialState,
    setPlatform
};
