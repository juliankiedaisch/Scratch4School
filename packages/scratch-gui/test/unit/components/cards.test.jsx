import React from 'react';
import {renderWithIntl} from '../../helpers/intl-helpers.jsx';
import '@testing-library/jest-dom';
import {PLATFORM} from '../../../src/lib/platform.js';

// Mock this utility because it uses dynamic imports that do not work with jest
jest.mock('../../../src/lib/libraries/decks/translate-image.js', () => { });

import Cards from '../../../src/components/cards/cards.jsx';

describe('Cards component', () => {
    const defaultProps = () => ({
        activeDeckId: 'id1',
        content: {
            id1: {
                name: 'id1 - name',
                img: 'id1 - img',
                steps: [{video: 'videoUrl'}]
            }
        },
        dragging: false,
        expanded: true,
        isRtl: false,
        locale: 'en',
        onActivateDeckFactory: jest.fn(),
        onCloseCards: jest.fn(),
        onDrag: jest.fn(),
        onEndDrag: jest.fn(),
        onNextStep: jest.fn(),
        onPrevStep: jest.fn(),
        onShowAll: jest.fn(),
        onShrinkExpandCards: jest.fn(),
        onStartDrag: jest.fn(),
        showVideos: true,
        step: 0,
        x: 0,
        y: 0
    });

    test('showVideos=true shows the video step', () => {
        const {container} = renderWithIntl(<Cards
            {...defaultProps()}
            platform={PLATFORM.WEB}
            showVideos
        />);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('showVideos=false shows the title image/name instead of video step', () => {
        const {container} = renderWithIntl(<Cards
            {...defaultProps()}
            showVideos={false}
        />);

        expect(container.firstChild).toMatchSnapshot();
    });
});
