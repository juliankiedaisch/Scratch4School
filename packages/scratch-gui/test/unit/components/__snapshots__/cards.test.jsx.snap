// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards component showVideos=false shows the title image/name instead of video step 1`] = `
<div
  style="width: 1824px; height: 977px; top: 48px; left: -400px;"
>
  <div
    class="react-draggable"
    style="transform: translate(692px,340px);"
  >
    <div>
      <div>
        <div>
          <img
            src="test-file-stub"
          />
          Tutorials
        </div>
        <div>
          <div>
            <img
              draggable="false"
              src="test-file-stub"
            />
            Shrink
          </div>
          <div>
            <img
              src="test-file-stub"
            />
            Close
          </div>
        </div>
      </div>
      <div>
        <div>
          id1 - name
        </div>
        <div>
          <img
            draggable="false"
            src="id1 - img"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Cards component showVideos=true shows the video step 1`] = `
<div
  style="width: 1824px; height: 977px; top: 48px; left: -400px;"
>
  <div
    class="react-draggable"
    style="transform: translate(692px,340px);"
  >
    <div>
      <div>
        <div>
          <img
            src="test-file-stub"
          />
          Tutorials
        </div>
        <div>
          <div>
            <img
              draggable="false"
              src="test-file-stub"
            />
            Shrink
          </div>
          <div>
            <img
              src="test-file-stub"
            />
            Close
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            class="wistia_embed wistia_async_videoUrl"
            id="video-div"
            style="height: 257px; width: 466px;"
          >
            Â 
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
